### ISSUE TITLE
Pre-releases and release candidates

### ISSUE THREAD

**Issue by brainsiq** [tjunnone/npm-check-updates#227]
I'm not sure if this is intended behaviour or not as I find the README slightly unclear on the usage of the `--greatest` flag. Am running npm-check-updates without this flag and it is trying to upgrade async to a release candidate and redis to a pre-release version.

`"async": "^1.5.2"` to `"async": "^2.0.0-rc.3"`
`"redis": "^2.5.3"` to `"redis": "^2.6.0-1"`

**Comment by brainsiq** [comment]
The updated versions are what are returned from `npm info` in dist-tags.latest so is it the package maintainers fault?

**Comment by metaraine** [comment]
Yes, if they published a release candidate to the latest tag they are saying that that is the latest stable release. They should publish to something other than latest if they want it to be treated as a prerelease.

**Comment by brainsiq** [comment]
NPM's guide on semver does state that appending `-x` indicates a prerelease version (https://docs.npmjs.com/misc/semver#prerelease-tags). 

Would you be opposed to parsing the version and excluding anything that looks like a prerelease with this format? Happy to get involved in helping to make this change.

**Comment by brainsiq** [comment]
Bit of relevant discussion in the redis repo: https://github.com/NodeRedis/node_redis/issues/1026

**Comment by metaraine** [comment]
The semver spec does indicate how pre-release versions should be handled in terms of satisfying version ranges, but this is separate from how npm handles the "latest" tag in its registry and corresponding install behavior. 

From my perspective it is pretty clear that latest should always point to the latest stable version and other tags can be used for pre-releases. (I think there should be a standardized tag like "next" but that's another issue). 

I would be opposed to custom parsing of the version number as a workaround for aberrant library versioning. 

My best recommendation would be to filter out the offending package and run a separate ncu command with the greatest tag just for that package:
ncu -x redis && ncu -t redis

**Comment by brainsiq** [comment]
Thanks for the detailed response. I can put up with ignoring these 2 for the time being.

**Comment by amannn** [comment]
I'm running into the same issue.

```
 react                    15.2.0  →            15.3.0-rc.2 
 react-dom                15.2.0  →            15.3.0-rc.2 
 babel-core                6.9.0  →                 6.10.4 
 babel-eslint              6.0.0  →                  6.1.2 
 babel-jest               12.1.0  →  12.1.5-alpha.b5322422 
 jest-cli                 12.1.0  →  12.1.5-alpha.b5322422 
 react-addons-test-utils  15.2.0  →            15.3.0-rc.2 
```

It might be tough to educate all of the authors about this issue, as it seems somewhat common.

Here's my temporary workaround:
```js
const ncu = require('npm-check-updates');

const commonConfig = {
  packageFile: './package.json',
  semverLevel: 'major'
};

ncu.run(Object.assign({}, commonConfig, {
  jsonUpgraded: true,
})).then(upgradeable => {
  // All pre-releases contain a dash; filter them out
  const deps = Object.keys(upgradeable).filter(key =>
    !upgradeable[key].includes('-'));
  ncu.run(Object.assign({}, commonConfig, {
    filter: deps.join(','),
    upgrade: true
  })).then(upgraded => {
    console.log(upgraded);
  });
});
```

The problem with this script is that if a dependency currently has its latest version set to a pre-release version, it won't update at all (even if there are stable releases inbetween).
